lib/rucio/core/permission/belleii.py

```diff
@@ -438,7 +407,26 @@ def perm_add_dids(issuer: "InternalAccount", kwargs: dict, *, session: "Optional
     """
     # Check the accounts of the issued rules
     for did in kwargs['dids']:
-        if not perm_add_did(issuer, kwargs=did, session=session):
+        # Check the accounts of the issued rules
+        for rule in did.get('rules', []):
+            did_rule = rule
+            if 'scope' not in did_rule:
+                if did['scope'] and not isinstance(did['scope'], str):
+                    did_rule['scope'] = did['scope'].external
+                else:
+                    did_rule['scope'] = did['scope']
+            if not perm_add_rule(issuer, kwargs=did_rule, session=session):
+                return False
+
+        scope = did['scope']
+        if isinstance(did['scope'], str):
+            scope = InternalScope(did['scope'])
+
+        if not perm_default(issuer, did, session=session)\
+                and not has_account_attribute(account=issuer, key='did_admin', session=session)\
+                and not has_account_attribute(account=issuer, key='production_account', session=session)\
+                and not rucio.core.scope.is_scope_owner(scope=scope, account=issuer, session=session)\
+                and not (did.get('name', False) and did['name'].startswith('/belle/scout')):
```

Comment:
I actually think that the approach that was there seems pretty good - maybe we should change the focus of the issue/PR into using this approach in the other permission files as well, so basically:

1. `perm_add_did` should have the logic to check permission for a single did
2. `perm_add_dids` should simply do `return all([perm_add_did(issuer, kwargs=did, session=session) for did in kwargs['dids']])`

and then we shouldn't have to change anything else other than this

Reply:
Fixed this in the latest force-push